tool/clean/clean_log_group/clean_log_group.go

```diff
@@ -0,0 +1,293 @@
+}
+
+const (
+	LogGroupProcessChanSize = 500
+)
+
+// Config holds the application configuration
+type Config struct {
+	creationThreshold time.Duration
+	inactiveThreshold time.Duration
+	numWorkers        int
+	deleteBatchCap    int
+	exceptionList     []string
+	dryRun            bool
+}
+
+// Global configuration
+var (
+	cfg Config
+)
+
+func init() {
+	// Set default configuration
+	cfg = Config{
+		creationThreshold: 3 * clean.KeepDurationOneDay,
+		inactiveThreshold: 1 * clean.KeepDurationOneDay,
+		numWorkers:        15,
+		exceptionList:     []string{"lambda"},
+		dryRun:            true,
+	}
+
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+	// Parse command line flags
+	flag.BoolVar(&cfg.dryRun, "dry-run", false, "Enable dry-run mode (no actual deletion)")
+	flag.Parse()
+	// Load AWS configuration
+	awsCfg, err := loadAWSConfig(ctx)
+	if err != nil {
+		log.Fatalf("Error loading AWS config: %v", err)
+	}
+
+	// Create CloudWatch Logs client
+	client := cloudwatchlogs.NewFromConfig(awsCfg)
+
+	// Compute cutoff times
+	cutoffTimes := calculateCutoffTimes()
+
+	log.Printf("🔍 Searching for CloudWatch Log Groups older than %d days AND inactive for %d days in %s region\n",
+		cfg.creationThreshold, cfg.inactiveThreshold, awsCfg.Region)
+
+	// Delete old log groups
+	deletedGroups := deleteOldLogGroups(ctx, client, cutoffTimes)
+	log.Printf("Total log groups deleted: %d", len(deletedGroups))
+}
+
+type cutoffTimes struct {
+	creation int64
+	inactive int64
+}
+
+func calculateCutoffTimes() cutoffTimes {
+	return cutoffTimes{
+		creation: time.Now().Add(cfg.creationThreshold).UnixMilli(),
+		inactive: time.Now().Add(cfg.inactiveThreshold).UnixMilli(),
+	}
+}
+
+func loadAWSConfig(ctx context.Context) (aws.Config, error) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return aws.Config{}, fmt.Errorf("loading AWS config: %w", err)
+	}
+	cfg.RetryMode = aws.RetryModeAdaptive
+	return cfg, nil
+}
+
+func deleteOldLogGroups(ctx context.Context, client cloudwatchlogsClient, times cutoffTimes) []string {
+	var (
+		wg                      sync.WaitGroup
+		deletedLogGroup         []string
+		foundLogGroupChan       = make(chan *types.LogGroup, LogGroupProcessChanSize)
+		deletedLogGroupNameChan = make(chan string, LogGroupProcessChanSize)
+		handlerWg               sync.WaitGroup
+	)
+
+	// Start worker pool
+	log.Printf("👷 Creating %d workers\n", cfg.numWorkers)
+	for i := 0; i < cfg.numWorkers; i++ {
+		wg.Add(1)
+		w := worker{
+			id:                   i,
+			wg:                   &wg,
+			incomingLogGroupChan: foundLogGroupChan,
+			deletedLogGroupChan:  deletedLogGroupNameChan,
+			times:                times,
+		}
+		go w.processLogGroup(ctx, client)
+	}
+
+	// Start handler with its own WaitGroup
+	handlerWg.Add(1)
+	go func() {
+		handleDeletedLogGroups(&deletedLogGroup, deletedLogGroupNameChan)
+		handlerWg.Done()
```

Comment:
nit: Typically use defer so it'll get called even if the `handleDeletedLogGroups` panics. Unlikely to happen, so it should be fine.
```suggestion
		defer handlerWg.Done()
```